internal/signaling/timeout_manager.go

```diff
@@ -88,6 +86,28 @@ func (i *TimeoutManager) Disconnected(ctx context.Context, p *Peer) {
 	err := i.Store.TimeoutPeer(ctx, p.ID, p.Secret, p.Game, lobbies)
 	if err != nil {
 		logger.Error("failed to record timeout peer", zap.Error(err))
+	} else {
```

Comment:
Do we maybe wanna do leaderElection anyway, even if the TimeoutPeer failed?

Reply:
If `TimeoutPeer` fails, the peer won't have been inserted into `timeouts` and `DoLeaderElection` will just keep it as the current leader if it is.

Reply:
And most likely if `TimeoutPeer` the database has issues and `DoLeaderElection` will also fail.